django/views/decorators/common.py

```diff
@@ -9,9 +11,19 @@ def no_append_slash(view_func):
 
     # view_func.should_append_slash = False would also work, but decorators are
     # nicer if they don't have side effects, so return a new function.
-    @wraps(view_func)
-    def wrapper_view(*args, **kwargs):
-        return view_func(*args, **kwargs)
 
-    wrapper_view.should_append_slash = False
-    return wrapper_view
+    if iscoroutinefunction(view_func):
+
+        async def _view_wrapper(request, *args, **kw):
+            response = await view_func(request, *args, **kw)
+            return response
```

Comment:
🤔 I wasn't sure if this would be neater than the single line option:

```py
return await view_func(request, *args, **kw)
```

I think the more verbose version makes the code a little clearer, but I'm not sure if one less variable assignment is a bigger benefit.

Reply:
Made it into a single line to keep the styling in line with other decorator styles.